daffodil-lib/src/main/scala/org/apache/daffodil/cookers/EntityReplacer.scala

```diff
@@ -818,20 +819,27 @@ class DelimiterCookerNoES(pn: String) extends ListOfString1OrMoreLiteral(pn, tru
   override val oneLiteralCooker: StringLiteralBase =
     new StringLiteralNoCharClassEntities(propName, true) with DisallowedCharClassEntitiesMixin {
 
-      //override protected def noCharClassEntities(raw: String, context: ThrowsSDE): Unit = {
-      // TODO: this isn't quite right, as it will allow combined delimiters
-      // that still match the empty string, e.g. "%ES;%WSP*;". We could check
-      // if raw.contains("%WSP*;"), but that is too general, preventing valid
-      // delimiters like "foo%WSP*;bar". Although the below matches the
-      // specification, it's probably not the intended behavior.
+      // Disallow "%ES" in the string raw. Disallow "%WSP*" when it is
+      // the only value in the string raw
       override val disallowedCharClassEntities = Seq("ES")
       override def testRaw(raw: String, context: ThrowsSDE) = {
         context.schemaDefinitionUnless(raw != "%WSP*;", """The WSP* entity cannot appear on it's own when dfdl:lengthKind="delimited".""")
         super[DisallowedCharClassEntitiesMixin].testRaw(raw, context)
       }
-      //context.schemaDefinitionUnless(raw != "%WSP*;", "Property dfdl:%s cannot contain %%WSP*; when dfdl:lengthKind=\"delimited\".", propName)
-      //context.schemaDefinitionUnless(raw != "%ES;", "Property dfdl:%s cannot contain %%ES; when dfdl:lengthKind=\"delimited\".", propName)
-      //}
+    }
+}
+
+class DelimiterCookerNoSoleES(pn: String) extends ListOfString1OrMoreLiteral(pn, true) {
+
+  override val oneLiteralCooker: StringLiteralBase =
+    new StringLiteralNoCharClassEntities(propName, true) {
+
+      // "Neither %ES or %WSP*;" can appear on their own when lenghtKind is delimited.
+      override def testRaw(raw: String, context: ThrowsSDE) = {
+        context.schemaDefinitionUnless(raw != "%ES;", """The ES entity cannot appear on its own when dfdl:lengthKind="delimited".""")
+        context.schemaDefinitionUnless(raw != "%WSP*;", """The WSP* entity cannot appear on its own when dfdl:lengthKind="delimited".""")
+        testRaw(raw, context)
```

Comment:
This is the line where things are probably hangning.

This line is calling ``testRaw``, which is itself, which creates an infinite loop and make things hang. Previously, this called ``super[DisallowedCharClassEntitiesMixin].testRaw``, which is a different ``testRaw`` function in a different trait that has additional logic for checking for disallowed entities. But this class has no disallowed entities, so we don't need to call it, or any other testRaw function. This function contains all the logic we need.

So removing this line should prevent the hang, and hopefully allow the tests to pass.